tests/ts/examples.ts

```diff
@@ -584,6 +593,107 @@ Person.query().where(ref('Model.jsonColumn:details'), '=', lit({ name: 'Jennifer
 Person.query().where('age', '>', lit(10));
 Person.query().where('firstName', lit('Jennifer').castText());
 
+// Preserving result type after result type changing methods.
+
+const findByIdSelect: Promise<Person | undefined> = Person.query()
+  .findById(32)
+  .select('firstName');
+const findByIdSelectThrow: Promise<Person> = Person.query()
+  .findById(32)
+  .select('firstName')
+  .throwIfNotFound();
+const findByIdJoin: Promise<Person | undefined> = Person.query()
+  .findById(32)
+  .join('tablename', 'column1', '=', 'column2');
+const findByIdJoinThrow: Promise<Person> = Person.query()
+  .findById(32)
+  .join('tablename', 'column1', '=', 'column2')
+  .throwIfNotFound();
+const findByIdJoinRaw: Promise<Person | undefined> = Person.query()
+  .findById(32)
+  .joinRaw('raw sql');
+const findByIdJoinRawThrow: Promise<Person> = Person.query()
+  .findById(32)
+  .joinRaw('raw sql')
+  .throwIfNotFound();
+const findOneWhere: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .where('more raw sql');
+const findOneWhereThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .where('more raw sql')
+  .throwIfNotFound();
+const findOneSelect: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .select('firstName');
+const findOneSelectThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .select('firstName')
+  .throwIfNotFound();
+const findOneWhereIn: Promise<Person | undefined> = Person.query()
+  .findOne('raw sql')
+  .whereIn('status', ['active', 'pending']);
+const findOneWhereInThrow: Promise<Person> = Person.query()
+  .findOne('raw sql')
+  .whereIn('status', ['active', 'pending'])
+  .throwIfNotFound();
```

Comment:
You cannot give a string for `findOne`. Also none of the methods accept raw sql as a string. Either `objection.raw` or `knex.raw` needs to be used.

Reply:
I don't know if it should be explicitly prevented, but having these examples is misleading.

Reply:
I'll change the `findOne()` parameters to something valid. They don't matter for the purpose of this test, so might as well pick something that's possible.